File: archive/Libs/MessageLib/cs/Messages.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace MessageLib
{
    public class Error : Message
    {
        public string Message { get; set; }

        public Error() { }

        public Error(string message) { this.Message = message; }
    }
}

File: archive/Libs/MessageLib/ts/Messages.ts
module MessageLib {

    export interface IMessage {
        $type?: string;
        Type: string; 
    }

    export interface IError extends IMessage {
        Message: string;
    }

    export class Message implements IMessage {
        $type: string;
        Type: string;
        constructor() {
            this.$type = this.Type = typeof this;
        }
    }

    export class Error extends Message implements IError {
        Message: string;
    }
}
